sdks/python/apache_beam/runners/worker/statesampler.py

```diff
@@ -62,12 +64,22 @@ def __init__(self, prefix, counter_factory,
                sampling_period_ms=DEFAULT_SAMPLING_PERIOD_MS):
     self.states_by_name = {}
     self._prefix = prefix
-    self._counter_factory = counter_factory
+    self._counter_factory = counter_factory or CounterFactory()
```

Comment:
It looks like the second branch is only used by tests.  Can we have the tests pass empty `CounterFactory()`s instead of adding this optional behavior in the actual code?